src/test/llmnode_test.cpp

```diff
@@ -3037,6 +3109,70 @@ TEST(EmbeddingsSerialization, multipleStringInput) {
     ASSERT_EQ(strings->at(2), "three");
 }
 
+TEST(EmbeddingsSerialization, intInput) {
+    std::string requestBody = R"(
+        {
+            "model": "embeddings",
+            "input": [1, 2, 3]
+        }
+    )";
+    rapidjson::Document d;
+    rapidjson::ParseResult ok = d.Parse(requestBody.c_str());
+    ASSERT_EQ(ok.Code(), 0);
+    auto request = EmbeddingsRequest::fromJson(&d);
+    ASSERT_EQ(std::get_if<std::string>(&request), nullptr);
+    auto embeddingsRequest = std::get<EmbeddingsRequest>(request);
+    ASSERT_EQ(embeddingsRequest.encoding_format, EncodingFormat::FLOAT);
+    auto ints = std::get_if<std::vector<std::vector<int>>>(&embeddingsRequest.input);
+    ASSERT_NE(ints, nullptr);
+    ASSERT_EQ(ints->size(), 1);
+    ASSERT_EQ(ints->at(0).size(), 3);
+    ASSERT_EQ(ints->at(0).at(0), 1);
+    ASSERT_EQ(ints->at(0).at(1), 2);
+    ASSERT_EQ(ints->at(0).at(2), 3);
+}
+
+TEST(EmbeddingsSerialization, multipleIntInput) {
+    std::string requestBody = R"(
+        {
+            "model": "embeddings",
+            "input": [[1, 2, 3], [4, 5, 6]]
```

Comment:
add tests with variable tokens length, we should confirm padding and attention mask is created correctly